modules/tinymce/src/plugins/table/main/ts/ui/Helpers.ts

```diff
@@ -267,17 +267,20 @@ export interface CellData {
   backgroundcolor?: string;
 }
 
-const extractDataFromCellElement = (editor: Editor, elm: HTMLElement, hasAdvancedCellTab: boolean): CellData => {
+const extractDataFromCellElement = (editor: Editor, cell: HTMLTableDataCellElement, column: SugarElement<HTMLTableColElement>, hasAdvancedCellTab: boolean): CellData => {
   const dom = editor.dom;
+
+  const getStyle = (element: HTMLElement, style: string) => dom.getStyle(element, style) || dom.getAttrib(element, style);
+
   return {
-    width: dom.getStyle(elm, 'width') || dom.getAttrib(elm, 'width'),
-    height: dom.getStyle(elm, 'height') || dom.getAttrib(elm, 'height'),
-    scope: dom.getAttrib(elm, 'scope'),
-    celltype: Util.getNodeName(elm),
-    class: dom.getAttrib(elm, 'class', ''),
-    halign: getHAlignment(editor, elm),
-    valign: getVAlignment(editor, elm),
-    ...(hasAdvancedCellTab ? extractAdvancedStyles(dom, elm) : {})
+    width: column ? getStyle(column.dom, 'width') : getStyle(cell, 'width'),
```

Comment:
Continuing from this, I'm okay with merging this as is for now but we do prefer to avoid using undefined, so we may want to change this later. Probably to an Optional.